src/server/network/protocol/protocolgame.cpp

```diff
@@ -6809,8 +6810,8 @@ void ProtocolGame::AddPlayerStats(NetworkMessage &msg) {
 	msg.add<uint16_t>(player->getStaminaXpBoost()); // stamina multiplier (100 = 1.0x)
 
 	if (!oldProtocol) {
-		msg.add<uint32_t>(std::min<int32_t>(player->getMana(), std::numeric_limits<uint16_t>::max()));
-		msg.add<uint32_t>(std::min<int32_t>(player->getMaxMana(), std::numeric_limits<uint16_t>::max()));
+		msg.add<uint32_t>(std::min<int32_t>(player->getMana(), 10000000));
```

Comment:
I believe the problem is that the base currently limits health to int32_t, which the maximum is less than the maximum of uint32_t. Have you tested sending the maximum of int32_t instead of sending a random hardcoded value?